### PR TITLE
Exemption Certificate

### PR DESCRIPTION
**Pull request by martypowell**
Addresses #32 

- Shows exemption form if an exemption value is present
- Add, Edit and Remove Exemptions
- Exemption List is added as a formik field

Sorry for the size of this one.

### CODE REVIEW

File: src/components/ExemptionCertificate.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    setExemptions(exemptions.filter(exemption => exemption.id !== exemptionId));
+  };
+
+  return (
+    <div className="tt_form-section">
+      <p>
+        I certify that the occupancy of the facilities above have been (or will
+        be) furnished for the exclusive use of and will be paid by or from the
+        funds of:
+      </p>
```
Comment (tmccoy529) [inline src/components/ExemptionCertificate.jsx]: Should this be a constant label?

File: src/components/ExemptionCertificate.jsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+          exemptions={exemptions}
+          handleEditClick={editExemption}
+          handleRemoveClick={removeExemption}
+        />
+      )}
+      <p>
+        That all of the qualifications explained on this form have been met so
+        as to exempt this occupancy from the tax imposed by County Code
+        11-4-401.
+      </p>
```
Comment (tmccoy529) [inline src/components/ExemptionCertificate.jsx]: See above.

File: src/components/forms/TransientTaxForm.jsx [hunk 3]
```diff
@@ -113,7 +125,7 @@ const TransientTaxForm = props => (
             <PaymentOptions />
             <ReturnDateSelector paymentInterval={paymentInterval} />
           </div>
-          {Object.keys(monthsToReport).length > 0 && (
+          {isPaymentIntervalSelected && (
             <React.Fragment>
```
Comment (tmccoy529) [inline src/components/forms/TransientTaxForm.jsx]: nice change here, makes it easier to follow

**Comment by tmccoy529** [comment]
image

Date error

**Comment by tmccoy529** [comment]
image

added exemption without selecting an option

File: src/components/ExemptionCertificate.jsx (hunk 1 above)
Comment (martypowell) [reply src/components/ExemptionCertificate.jsx]: @tmccoy529 yes I think that probably makes sense.

File: src/components/ExemptionCertificate.jsx (hunk 2 above)
Comment (martypowell) [reply src/components/ExemptionCertificate.jsx]: @tmccoy529 yes I think that probably makes sense.

File: src/components/ExemptionCertificate.jsx (hunk 1 above)
Comment (martypowell) [reply src/components/ExemptionCertificate.jsx]: @tmccoy529 let me know if what i did makes sense.

- Created a Messages object in constants
- Created a specific section for the Exemption Cert
- Used those in the form

File: src/components/ExemptionCertificate.jsx (hunk 2 above)
Comment (martypowell) [reply src/components/ExemptionCertificate.jsx]: See https://github.com/baltimorecounty/react-transient-tax/pull/58#discussion_r347361622

**Comment by martypowell** [comment]
> image
> 
> added exemption without selecting an option

@tmccoy529 all validation addressed.

- Handled outside of formik for now, since it was quicker
- Displays a list of errors above the exemption selector form
- Will not allow user to submit without type, fromDate, toDate

One note, we will probably need to style this, but i didn't want to get to far in the weeds so I just added red color to the list for now.

**Comment by martypowell** [comment]
@sgrg1 brought up a good point, about when a user edits an exemption but does not save cahnges. I have added an [enhancement](#60) to address this.

**Pull request merged**